hmpps-prisoner-search-indexer/README.md

```diff
@@ -0,0 +1,550 @@
+# HMPPS Prisoner Search Indexer
+
+[![CircleCI](https://circleci.com/gh/ministryofjustice/hmpps-prisoner-search/tree/main.svg?style=svg)](https://circleci.com/gh/ministryofjustice/hmpps-prisoner-search-indexer)
+[![Docker Repository on Quay](https://quay.io/repository/hmpps/hmpps-prisoner-search-indexer/status "Docker Repository on Quay")](https://quay.io/repository/hmpps/hmpps-prisoner-search-indexer)
+[![API docs](https://img.shields.io/badge/API_docs_-view-85EA2D.svg?logo=swagger)](https://prisoner-search-indexer-dev.prison.service.justice.gov.uk/swagger-ui/index.html?configUrl=/v3/api-docs)
+
+The purpose of this service is to:
+* Keep the OpenSearch prison index up to date with changes from Prison systems (NOMIS)
+* Rebuild the index when required without an outage
+
+## Prisoner updates
+
+This service subscribes to both domain events and prison offender events.
+
+When an event is received a message is put onto the event queue.  The event queue then processes that message -
+the latest prisoner record is retrieved via the `prison-api`, augmented with incentives and restricted patient data 
+and upserted into the prisoner index.
+If the message processing fails then the message is transferred onto the event dead letter queue (DLQ).
+
+## Index rebuilds
+
+This service maintains two indexes `prisoner-search-green` and `prisoner-search-blue` know in the code as `GREEN` and `BLUE`.
+
+In normal running one of these indexes will be "active" while the other is dormant and not in use.
+
+When we are ready to rebuild the index the "other" non-active index is transitioned into a `BUILDING` state of `true`.
+
+```shell
+    http PUT /maintain-index/build
+```
+
+The entire NOMIS prisoner base is retrieved and over several hours the other index is fully populated.
+
+Once the index has finished, if there are no errors then the (housekeeping cronjob)[#housekeeping-cronjob] will mark the
+index as complete and switch to the new index.
+
+If the index build fails - there are messages left on the index dead letter queue - then the new index will remain 
+inactive until the DLQ is empty. It may take user intervention to clear the DLQ if some messages are genuinely 
+unprocessable (rather than just failed due to e.g. network issues).
+
+#### Index switch
```

Comment:
only reached this far today before I lost the plot.  Will carry on with the rest next week but thought might as well commit what I've done so far

Reply:
below this is verbatim copied from the old search so needs amending